tests/pjit_test.py

```diff
@@ -4113,6 +4113,22 @@ def f():
     with mesh:
       f()  # doesn't crash
 
+  def test_lowering_cache_hit_with_closed_over_constants(self):
```

Comment:
Would it be worth having a test where we are closing over a numpy_array in a nested jit jaxpr and we can see that it's being converted to an arg now but before it was embedded as a constant?